src/pretix/control/views/auth.py

```diff
@@ -427,25 +431,41 @@ def post(self, request, *args, **kwargs):
                 devices = U2FDevice.objects.filter(user=self.user)
 
             for d in devices:
+                credential_current_sign_count = d.sign_count if isinstance(d, WebAuthnDevice) else 0
                 try:
-                    wu = d.webauthnuser
-
-                    if isinstance(d, U2FDevice):
-                        # RP_ID needs to be appId for U2F devices, but we can't
-                        # set it that way in U2FDevice.webauthnuser, since that
-                        # breaks the frontend part.
-                        wu.rp_id = settings.SITE_URL
-
-                    webauthn_assertion_response = webauthn.WebAuthnAssertionResponse(
-                        wu,
-                        resp,
-                        challenge,
-                        settings.SITE_URL,
-                        uv_required=False  # User Verification
+                    webauthn_assertion_response = webauthn.verify_authentication_response(
+                        credential=resp,
+                        expected_challenge=base64.b64decode(challenge),
+                        expected_rp_id=get_webauthn_rp_id(self.request),
+                        expected_origin=settings.SITE_URL.rstrip("/"),
```

Comment:
Is the `.rstrip("/")` just a safe-guard as https://github.com/pretix/pretix/blob/master/src/pretix/settings.py#L189 should have already removed it.

```suggestion
                        expected_origin=settings.SITE_URL,
```